File: 高级数据结构与算法/Python/第4章 模式匹配/4.1 BF/brute_force.py
def brute_force(s, p):
    s_len = len(s)
    p_len = len(p)
    i = 0
    j = 0

    while i < s_len and j < p_len:
        if s[i] == p[j]:
            i += 1
            j += 1
        else:
            i = i - j + 1
            j = 0
    
    if j == p_len:
        return i - j
    else:
        return -1

def main():
    s = "BBC ABCDAB ABCDABCDABDE"
    p = "ABCDABD"
    print(brute_force(s, p))

if __name__ == "__main__":
    main()

File: code/Java/Chapter13/13-1/BruteForce.java
public class BruteForce {
    public static void main(String[] args) {
        String s = "BBC ABCDAB ABCDABCDABDE";
        String p = "ABCDABD";
        System.out.println(bruteForce(s, p));
    }

    public static int bruteForce(String S, String P) {
        char[] s = S.toCharArray();
        char[] p = P.toCharArray();
        int i = 0;
        int j = 0;

        while (i < s.length && j < p.length) {
            if (s[i] == p[j]) {
                i++;
                j++;
            } else {
                i = i - j + 1;
                j = 0;
            }
        }

        if (j == p.length) {
            return i - j;
        } else {
            return -1;
        }
    }
}